SPHINXsys/src/shared/particle_dynamics/solid_dynamics/contact_dynamics.cpp

```diff
@@ -345,96 +336,32 @@ namespace SPH
-												*solid_body_contact_relation.body_surface_layer_),
-			  ContactDynamicsData(solid_body_contact_relation),
-			  Vol_(particles_->Vol_), mass_(particles_->mass_),
-			  vel_n_(particles_->vel_n_), dvel_dt_prior_(particles_->dvel_dt_prior_),
-			  contact_force_(particles_->contact_force_), penalty_strength_(penalty_strength)
-		{
-			Real impedance = material_->ReferenceDensity() * sqrt(material_->ContactStiffness());
-			Real reference_pressure = material_->ReferenceDensity() * material_->ContactStiffness();
-			for (size_t k = 0; k != contact_particles_.size(); ++k)
-			{
-				contact_Vol_.push_back(&(contact_particles_[k]->Vol_));
-				contact_vel_n_.push_back(&(contact_particles_[k]->vel_n_));
-				Real contact_impedance =
-					contact_material_[k]->ReferenceDensity() * sqrt(contact_material_[k]->ContactStiffness());
-				contact_impedance_.push_back(2.0 * impedance * contact_impedance / (impedance + contact_impedance));
-				Real contact_reference_pressure =
-					contact_material_[k]->ReferenceDensity() * contact_material_[k]->ContactStiffness();
-				contact_reference_pressure_.push_back(2.0 * reference_pressure * contact_reference_pressure /
-													  (reference_pressure + contact_reference_pressure));
 			}
 		}
 		//=================================================================================================//
-		void DynamicContactForce::Interaction(size_t index_i, Real dt)
-		{
-			Real Vol_i = Vol_[index_i];
-			Vecd vel_i = vel_n_[index_i];
-
-			/** Contact interaction. */
-			Vecd force(0.0);
-			for (size_t k = 0; k < contact_configuration_.size(); ++k)
-			{
-				Real particle_spacing_j1 = 1.0 / contact_bodies_[k]->sph_adaptation_->ReferenceSpacing();
-				Real particle_spacing_ratio2 =
-					1.0 / (this->body_->sph_adaptation_->ReferenceSpacing() * particle_spacing_j1);
-				particle_spacing_ratio2 *= 0.1 * particle_spacing_ratio2;
-
-				StdLargeVec<Real> &Vol_k = *(contact_Vol_[k]);
-				StdLargeVec<Vecd> &vel_n_k = *(contact_vel_n_[k]);
-
-				Neighborhood &contact_neighborhood = (*contact_configuration_[k])[index_i];
-				for (size_t n = 0; n != contact_neighborhood.current_size_; ++n)
-				{
-					size_t index_j = contact_neighborhood.j_[n];
-					Vecd e_ij = contact_neighborhood.e_ij_[n];
-
-					Real impedance_p = 0.5 * contact_impedance_[k] * (SimTK::dot(vel_i - vel_n_k[index_j], -e_ij));
-					Real overlap = contact_neighborhood.r_ij_[n];
-					Real delta = 2.0 * overlap * particle_spacing_j1;
-					Real beta = delta < 1.0 ? (1.0 - delta) * (1.0 - delta) * particle_spacing_ratio2 : 0.0;
-					Real penalty_p = penalty_strength_ * beta * overlap * contact_reference_pressure_[k];
-
-					// force due to pressure
-					force -= 2.0 * (impedance_p + penalty_p) * e_ij *
-							 Vol_i * Vol_k[index_j] * contact_neighborhood.dW_ij_[n];
-				}
-			}
-
-			contact_force_[index_i] = force;
-			dvel_dt_prior_[index_i] += force / mass_[index_i];
-		}
```

Comment:
Why is it getting removed ?

Reply:
It was used when i suppose the ordinary contact force model is not OK dynamical cases. However, it is found it is works too. so I think this no use now.